node/src/components/block_accumulator/tests.rs

```diff
@@ -1901,3 +1904,190 @@ async fn block_accumulator_reactor_flow() {
+        );
+        validator_matrix.register_era_validator_weights(weights);
+    }
+
+    // Register a signature for block 1.
+    {
+        let effect_builder = runner.effect_builder();
+        let reactor = runner.reactor_mut();
+
+        let block_accumulator = &mut reactor.block_accumulator;
+        block_accumulator.register_local_tip(0, 0.into());
+
+        let event = super::Event::ReceivedFinalitySignature {
+            finality_signature: Box::new(fin_sig_bob.clone()),
+            sender: peer_1,
+        };
+        let effects = block_accumulator.handle_event(effect_builder, &mut rng, event);
+        assert!(effects.is_empty());
+    }
+
+    // Register a signature for block 1.
+    {
+        let effect_builder = runner.effect_builder();
+        let reactor = runner.reactor_mut();
+
+        let block_accumulator = &mut reactor.block_accumulator;
+
+        let event = super::Event::ReceivedFinalitySignature {
+            finality_signature: Box::new(fin_sig_carol.clone()),
+            sender: peer_1,
+        };
+        let effects = block_accumulator.handle_event(effect_builder, &mut rng, event);
+        assert!(effects.is_empty());
+    }
+
+    // Register a signature for block 1.
+    {
+        let effect_builder = runner.effect_builder();
+        let reactor = runner.reactor_mut();
+
+        let block_accumulator = &mut reactor.block_accumulator;
+
+        let event = super::Event::CreatedFinalitySignature {
+            finality_signature: Box::new(fin_sig_alice.clone()),
+        };
+        let effects = block_accumulator.handle_event(effect_builder, &mut rng, event);
+        assert!(effects.is_empty());
+    }
+
+    // Register block 1.
+    {
+        runner
+            .process_injected_effects(|effect_builder| {
+                let event = super::Event::ReceivedBlock {
+                    block: Arc::new(block_1.clone()),
+                    sender: peer_2,
+                };
+                effect_builder
+                    .into_inner()
+                    .schedule(event, QueueKind::Validation)
+                    .ignore()
+            })
+            .await;
+        for _ in 0..6 {
+            while runner.try_crank(&mut rng).await == TryCrankOutcome::NoEventsToProcess {
+                time::sleep(POLL_INTERVAL).await;
+            }
+        }
+        let expected_block = runner
+            .reactor()
+            .storage
+            .read_block(block_1.hash())
+            .unwrap()
+            .unwrap();
+        assert_eq!(expected_block, block_1);
+        let expected_block_signatures = runner
+            .reactor()
+            .storage
+            .get_finality_signatures_for_block(*block_1.hash());
+        assert_eq!(
+            expected_block_signatures
+                .and_then(|sigs| sigs.get_finality_signature(&fin_sig_alice.public_key))
+                .unwrap(),
+            fin_sig_alice
+        );
+    }
+
+    tokio::time::sleep(
```

Comment:
I would also consider explaining this line, since it's actually crucial to the test.
Especially taking into consideration that we have comments for much more self-explanatory stuff like "Create 1 block." :-)

Reply:
Added in https://github.com/casper-network/casper-node/pull/3783/commits/5cb5b887afcefd75958bec58ffa70e0336246739